File: src/services/reportingApi.ts
import { authFetch } from '../utils/authFetch';
/**
 * reportingApi.ts
 * Frontend client for reporting-service (port 4004).
 */

const BASE = '/api/reporting';

export const reportingApi = {
  async getActivityLogs(): Promise<{ logs: any[] }> {
    const res = await authFetch(`${BASE}/activity-logs`);
    return res.json();
  },

  async logActivity(payload: {
    userId: string;
    userEmail: string;
    actionType: string;
    actionDetails: string;
    entityType: string;
    entityId: string | null;
  }) {
    const res = await authFetch(`${BASE}/activity-logs`, {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(payload),
    });
    return res.json();
  },

  async getShiftRecords(): Promise<{ records: any[] }> {
    const res = await authFetch(`${BASE}/shift-records`);
    return res.json();
  },
};

File: pos-android/app/src/main/java/com/pharmacare/pos/data/api/ReportingApi.kt
package com.pharmacare.pos.data.api

import retrofit2.Response
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Query

data class ActivityLogRequest(
    val userId: String,
    val userEmail: String,
    val actionType: String,
    val actionDetails: String,
    val entityType: String,
    val entityId: String? = null
)

interface ReportingApi {
    @POST("/api/reporting/activity-logs")
    suspend fun logActivity(@Body body: ActivityLogRequest): Response<Map<String, Any>>

    @GET("/api/reporting/activity-logs")
    suspend fun getActivityLogs(): Response<Map<String, Any>>

    @GET("/api/reporting/shift-records")
    suspend fun getShiftRecords(): Response<Map<String, Any>>
}
